x-pack/legacy/plugins/siem/public/components/open_timeline/open_timeline_modal/index.tsx

```diff
@@ -20,90 +19,61 @@ export interface OpenTimelineModalButtonProps {
   onToggle?: () => void;
 }
 
-export interface OpenTimelineModalButtonState {
-  showModal: boolean;
-}
-
 const DEFAULT_SEARCH_RESULTS_PER_PAGE = 10;
 const OPEN_TIMELINE_MODAL_WIDTH = 1000; // px
 
-// TODO: this container can be removed when
-// the following EUI PR is available (in Kibana):
-// https://github.com/elastic/eui/pull/1902/files#diff-d662c14c5dcd7e4b41028bf60b9bc77b
-const ModalContainer = styled.div`
-  .euiModalBody {
-    display: flex;
-    flex-direction: column;
-  }
-`;
-
-ModalContainer.displayName = 'ModalContainer';
-
 /**
  * Renders a button that when clicked, displays the `Open Timelines` modal
  */
-export class OpenTimelineModalButton extends React.PureComponent<
-  OpenTimelineModalButtonProps,
-  OpenTimelineModalButtonState
-> {
-  constructor(props: OpenTimelineModalButtonProps) {
-    super(props);
+export const OpenTimelineModalButton = React.memo<OpenTimelineModalButtonProps>(({ onToggle }) => {
+  const [showModal, setShowModal] = useState<boolean>(false);
 
-    this.state = { showModal: false };
+  /** shows or hides the `Open Timeline` modal */
+  function toggleShowModal() {
+    if (onToggle != null) {
+      onToggle();
+    }
+    setShowModal(!showModal);
   }
 
-  public render() {
-    return (
-      <ApolloConsumer>
-        {client => (
-          <>
-            <EuiButtonEmpty
-              color="text"
-              data-test-subj="open-timeline-button"
-              iconSide="left"
-              iconType="folderOpen"
-              onClick={this.toggleShowModal}
-            >
-              {i18n.OPEN_TIMELINE}
-            </EuiButtonEmpty>
-
-            {this.state.showModal && (
-              <EuiOverlayMask>
-                <ModalContainer>
-                  <EuiModal
-                    data-test-subj="open-timeline-modal"
-                    maxWidth={OPEN_TIMELINE_MODAL_WIDTH}
-                    onClose={this.toggleShowModal}
-                  >
-                    <StatefulOpenTimeline
-                      apolloClient={client}
-                      closeModalTimeline={this.closeModalTimeline}
-                      isModal={true}
-                      defaultPageSize={DEFAULT_SEARCH_RESULTS_PER_PAGE}
-                      title={i18n.OPEN_TIMELINE_TITLE}
-                    />
-                  </EuiModal>
-                </ModalContainer>
-              </EuiOverlayMask>
-            )}
-          </>
-        )}
-      </ApolloConsumer>
-    );
+  function closeModalTimeline() {
+    toggleShowModal();
   }
+  return (
+    <ApolloConsumer>
```

Comment:
instead of using `ApolloConsumer` here we can use the hookd `useApolloClient` from `x-pack/legacy/plugins/siem/public/utils/apollo_context.ts`

Reply:
would that require refactoring `open_timeline`? Possibly a new issue, or a part of this one which tackles all the `Query` components: https://github.com/elastic/kibana/issues/46261